src/core/image.rs

```diff
@@ -106,6 +83,114 @@ where
     }
 }
 
+#[derive(Debug)]
+pub struct RunnableImage<I: Image> {
+    image: I,
+    image_args: I::Args,
+    image_tag: Option<String>,
```

Comment:
I am kinda neutral in whether `image_tag` should be here or not. WDYT?

Reply:
Am I missing something or is it currently unused? Overriding the tag should be something that we can provide generically so yes, I think this should be in here.

Can you add docs to the `with_tag` function that explains that once that is used, there are no more guarantees that the container will actually work?

Reply:
Actually, based on what you think, I was intending to discuss this: So, right now, few images have `with_tag` as a helper function in their image implementations. How should we go about them? Can we delete them and use `with_tag` from `RunnableImage`?

Reply:
Yes we can do that !